### PR TITLE
Add com.leohanney.Volaris

### PR DESCRIPTION
**Pull request by greendoescode**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.
Volaris is an encryption / decryption tool, Built using Rust.

- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read the and followed all the [Submission and licence requirements][reqs].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by greendoescode** [comment]
bot, build com.leohanney.Volaris

File: com.leohanney.Volaris.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  - --share=ipc
+
+          
+modules:
+  - name: binary
+    buildsystem: simple
+    sources:
+      - type: file
+        url: https://github.com/volar-is/volaris-gui/releases/download/app-v0.1.0/Volaris_0.1.0_amd64.deb
+        sha256: c55febeee6cbde904b52119e86606dd08eb474aa8fc53be16843841334cdcc3c
```
Comment (bbhtt) [inline com.leohanney.Volaris.yml:21]: why not build from source ?

File: com.leohanney.Volaris.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+id: com.leohanney.Volaris
+runtime: org.gnome.Platform
+runtime-version: '46'
```
Comment (bbhtt) [inline com.leohanney.Volaris.yml:3]:
```suggestion
runtime-version: '47'
```

47 is the latest

File: com.leohanney.Volaris.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        url: https://raw.githubusercontent.com/volar-is/volaris-gui/main/flatpak.xml
+        sha256: 61ce16a64ab7afb0221f0d943bd381817a6b5617e993106003ca899fa9ede748
+                  
+    build-commands:
+       - ar -x *.deb
+       - tar -xf data.tar.gz
+       - mkdir -p /app/bin
+       - cp -r usr/bin/* /app/bin
+       - sed -i 's/Categories=/Categories=Utility/' usr/share/applications/volaris-gui.desktop
+       - sed -i 's#Comment=A Tauri App#Comment=Volaris GUI is a sleek Svelte/Tauri-based app for easy file encryption and decryption using key files.#' usr/share/applications/volaris-gui.desktop
```
Comment (bbhtt) [inline com.leohanney.Volaris.yml:33]: use patches or desktop-file-edit instead of sed

File: com.leohanney.Volaris.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+modules:
+  - name: binary
+    buildsystem: simple
+    sources:
+      - type: file
+        url: https://github.com/volar-is/volaris-gui/releases/download/app-v0.1.0/Volaris_0.1.0_amd64.deb
+        sha256: c55febeee6cbde904b52119e86606dd08eb474aa8fc53be16843841334cdcc3c
+        only-arches: [x86_64]
+      - type: file
+        url: https://raw.githubusercontent.com/volar-is/volaris-gui/main/flatpak.xml
```
Comment (bbhtt) [inline com.leohanney.Volaris.yml:24]: this needs to be from a commit not `main`.

File: com.leohanney.Volaris.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
+id: com.leohanney.Volaris
```
Comment (bbhtt) [inline com.leohanney.Volaris.yml:1]: can you link the github somewhere on https://leohanney.com/about ?

File: flathub.json [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
+    "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: this is disappointing.

File: com.leohanney.Volaris.yml (hunk 1 above)
Comment (greendoescode) [reply com.leohanney.Volaris.yml:21]: This is the official method suggested by Tauri

File: com.leohanney.Volaris.yml (hunk 4 above)
Comment (hfiguiere) [reply com.leohanney.Volaris.yml:24]: why isn't it in the .deb? it's clearly misnamed.

File: com.leohanney.Volaris.yml (hunk 3 above)
Comment (greendoescode) [reply com.leohanney.Volaris.yml:33]: noted, will do that

File: flathub.json (hunk 6 above)
Comment (greendoescode) [reply flathub.json:2]: I can't test arm support, hence I dont really want to included a build that might not work

File: com.leohanney.Volaris.yml (hunk 3 above)
Comment (hfiguiere) [reply com.leohanney.Volaris.yml:33]: note that "Svelte/Tauri" has nothing to do in that field. This is technobabble no user care about.

File: com.leohanney.Volaris.yml (hunk 1 above)
Comment (bbhtt) [reply com.leohanney.Volaris.yml:21]: they are suggesting it wrong, if it's possible please build from source. We have multiple tauri apps building from source on Flathub

File: com.leohanney.Volaris.yml (hunk 5 above)
Comment (greendoescode) [reply com.leohanney.Volaris.yml:1]: I could, it is all linked on volaris.leohanney.com, but im guessing thats not good enough?

File: com.leohanney.Volaris.yml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        only-arches: [x86_64]
+      - type: file
+        url: https://raw.githubusercontent.com/volar-is/volaris-gui/main/flatpak.xml
+        sha256: 61ce16a64ab7afb0221f0d943bd381817a6b5617e993106003ca899fa9ede748
+                  
+    build-commands:
+       - ar -x *.deb
+       - tar -xf data.tar.gz
+       - mkdir -p /app/bin
+       - cp -r usr/bin/* /app/bin
```
Comment (hfiguiere) [inline com.leohanney.Volaris.yml:31]: you could just `mv usr/* /app/` and then use the `rename-` in the manifest to fix icons and desktop file names automatically.

(but building from source would be better.)

File: flathub.json (hunk 6 above)
Comment (greendoescode) [reply flathub.json:2]: I'll remove this then

File: com.leohanney.Volaris.yml (hunk 1 above)
Comment (greendoescode) [reply com.leohanney.Volaris.yml:21]: > they are suggesting it wrong, if it's possible please build from source. We have multiple tauri apps building from source on Flathub

doing that now, thanks

File: com.leohanney.Volaris.yml (hunk 3 above)
Comment (greendoescode) [reply com.leohanney.Volaris.yml]: removed that

File: com.leohanney.Volaris.yml (hunk 5 above)
Comment (greendoescode) [reply com.leohanney.Volaris.yml:1]: https://leohanney.com/ would this new landing page work?

File: com.leohanney.Volaris.yml (hunk 1 above)
Comment (bbhtt) [reply com.leohanney.Volaris.yml:21]: if you can't then revert to using the DEBs.

File: com.leohanney.Volaris.yml (hunk 5 above)
Comment (bbhtt) [reply com.leohanney.Volaris.yml:1]: yes thanks

File: com.leohanney.Volaris.yml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+          
+modules:
+  - name: binary
+    buildsystem: simple
+    sources:
+      - type: file
+        url: https://github.com/volar-is/volaris-gui/releases/download/app-v0.1.0/Volaris_0.1.0_amd64.deb
+        sha256: c55febeee6cbde904b52119e86606dd08eb474aa8fc53be16843841334cdcc3c
+        only-arches: [x86_64]
```
Comment (bbhtt) [inline com.leohanney.Volaris.yml:22]:
```suggestion
        only-arches: [x86_64]
        dest-filename: Volaris.deb
```

File: com.leohanney.Volaris.yml [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+      - type: file
+        url: https://raw.githubusercontent.com/volar-is/volaris-gui/ff6e17fed31d868612a7a30805a3ed95f8552c21/flatpak.xml
+        sha256: 61ce16a64ab7afb0221f0d943bd381817a6b5617e993106003ca899fa9ede748
+                  
+    build-commands:
+       - ar -x *.deb
```
Comment (bbhtt) [inline com.leohanney.Volaris.yml:28]:
```suggestion
       - ar -x Volaris.deb
```

File: com.leohanney.Volaris.yml (hunk 1 above)
Comment (greendoescode) [reply com.leohanney.Volaris.yml:21]: > if you can't then revert to using the DEBs.

Alrighty, i'll stil try and get it to work in the background but having no luck with npm caches rn

**Comment by greendoescode** [comment]
bot, build com.leohanney.Volaris

**Comment by greendoescode** [comment]
Ah yeah this is gonna fail, no arm files yet, tryna figure that out rn

**Comment by greendoescode** [comment]
Seems like its a pita to set up to build locally so ill have to come back to that tommorow or another day to add arm support (CIs seem the only option)

**Comment by bbhtt** [comment]
> Ah yeah this is gonna fail, no arm files yet, tryna figure that out rn

https://docs.flathub.org/docs/for-app-authors/requirements#flathubjson if you can't build for aarch64

**Comment by greendoescode** [comment]
bot, build com.leohanney.Volaris

**Comment by greendoescode** [comment]
tested and it's working as expected

File: com.leohanney.Volaris.yml [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+runtime-version: '47'
+sdk: org.gnome.Sdk
+          
+command: volaris-gui
+rename-icon: com.leohanney.Volaris
+finish-args:
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --share=network
+  - --share=ipc
```
Comment (bbhtt) [inline com.leohanney.Volaris.yml:12]:
```suggestion
  - --share=network
  - --share=ipc
  - --device=dri
```

**Comment by greendoescode** [comment]
woop

**Pull request closed without merging**